tests/templates/test_subroutines/test_select.py

```diff
@@ -119,15 +171,103 @@ def circuit2():
+        ("ops", "control", "expected_gates", "wires"),
         [
             (
-                [qml.PauliX(wires=0), qml.PauliY(wires=0)],
+                [qml.X(0), qml.Y(0)],
                 [1],
                 [
-                    qml.ctrl(qml.PauliX(wires=0), control=1, control_values=0),
-                    qml.ctrl(qml.PauliY(wires=0), control=1),
+                    qml.ctrl(qml.X(0), control=1, control_values=0),
+                    qml.ctrl(qml.Y(0), control=1),
+                ],
+                2,
+            ),
+            (
+                [qml.X(0), qml.I(0), qml.Z(0)],
+                [1, 2],
+                [
+                    qml.ctrl(qml.X(0), control=[1, 2], control_values=[0, 0]),
+                    qml.ctrl(qml.Z(0), control=[1], control_values=[1]),
+                ],
+                3,
+            ),
+            (
+                [
+                    qml.X(0),
+                    qml.I(0),
+                    qml.I(0),
+                    qml.RX(0.3, 0),
+                ],
+                [1, 2],
+                [
+                    qml.ctrl(qml.X(0), control=[1, 2], control_values=[0, 0]),
+                    qml.ctrl(qml.RX(0.3, 0), control=[1, 2], control_values=[1, 1]),
+                ],
+                3,
+            ),
+            (
+                [qml.X("a"), qml.Z("b"), qml.RX(0.7, "b")],
+                ["c", 1],
+                [
+                    qml.ctrl(qml.X("a"), control=["c", 1], control_values=[0, 0]),
+                    qml.ctrl(qml.Z("b"), control=["c", 1], control_values=[0, 1]),
+                    qml.ctrl(qml.RX(0.7, "b"), control=["c"], control_values=[1]),
                 ],
+                ["a", "b", "c", 1],
+            ),
+            (
+                [qml.X("a"), qml.RX(0.7, "b")],
+                ["c", 1],
+                [
+                    qml.ctrl(qml.X("a"), control=[1], control_values=[0]),
+                    qml.ctrl(qml.RX(0.7, "b"), control=[1], control_values=[1]),
+                ],
+                ["a", "b", "c", 1],
+            ),
+        ],
+    )
+    def test_operation_result_partial(self, ops, control, expected_gates, wires, seed):
+        """Test the correctness of the Select template output with partial=True."""
+        dev = qml.device("default.qubit", wires=wires)
+
+        # Prepare a state that only has overlap with the basis states used in Select
+        np.random.seed(seed)
+        state = np.random.random(len(ops))
+        state /= np.linalg.norm(state)
+        state = np.concatenate([state, np.zeros(2 ** len(control) - len(ops))])
+
+        @qml.qnode(dev)
+        def circuit1():
+            qml.StatePrep(state, wires=control)
+            qml.Select(ops, control, partial=True)
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit2():
+            qml.StatePrep(state, wires=control)
+            for op in expected_gates:
+                qml.apply(op)
```

Comment:
You could also compare Select(..., partial = True) with Select(..., partial = False)